### ISSUE TITLE
YT-DLP on Heroku Giving m3u8 Instead of mp4 Downloadable Link. 

### ISSUE THREAD

**Issue by mischano** [yt-dlp/yt-dlp#7573]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I have a website that runs on `node.js` and `express` on the back end that in turn calls `.py` script to download a user-requested audio using yt-dlp. When I run the website on the localhost, everything runs perfectly, and I get a .mp4 downloadable url that can be fed right into the JavaScript audio.

However, when I deploy my website on Heroku, the same `.py` script gives me a .m3u8 url which is an audio playlist. 

I'm not sure if I'm extracting the url incorrectly with `extract_info` or if my `ffmpeg` is not working the right way in Heroku. Below is my `.py` script. Any help would be appreciated. Thanks. 

`ydl_opts = {
'format': 'bestaudio/best',
'quiet': True,
'postprocessors': [{
    'key': 'FFmpegExtractAudio',
    'preferredcodec': 'mp3',
    'preferredquality': '192',
}],}

with yt_dlp.YoutubeDL(ydl_opts) as ydl:
try:
    info = ydl.extract_info("ytsearch:%s" % requestedAudio, download=False)['entries'][0]
    # code follows...
except yt_dlp.utils.DownloadError or yt_dlp.utils.ExtractorError:
    # code follows...`

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
-----> Building on the Heroku-22 stack
-----> Using buildpacks:
       1. https://github.com/jonathanong/heroku-buildpack-ffmpeg-latest.git
       2. https://github.com/heroku/heroku-buildpack-nodejs#latest
       3. https://github.com/heroku/heroku-buildpack-python.git
-----> ffmpeg app detected
-----> Installing ffmpeg
       Variable FFMPEG_DOWNLOAD_URL isn't set, using default value
       Downloading https://johnvansickle.com/ffmpeg/builds/ffmpeg-git-amd64-static.tar.xz
       Unpacking the archive
       Installation successful
-----> Node.js app detected
       
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NODE_VERBOSE=false
       NODE_ENV=production
       NODE_MODULES_CACHE=true
       
-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)
       
       Resolving node version 18.x...
       Downloading and installing node 18.16.1...
       Using default npm version: 9.5.1
       
-----> Restoring cache
       - npm cache
       
-----> Installing dependencies
       Installing node modules
       
       added 117 packages, and audited 118 packages in 2s
       
       10 packages are looking for funding
         run `npm fund` for details
       
       3 moderate severity vulnerabilities
       
       To address all issues (including breaking changes), run:
         npm audit fix --force
       
       Run `npm audit` for details.
       
-----> Build
       
-----> Caching build
       - npm cache
       
-----> Pruning devDependencies
       
       up to date, audited 87 packages in 307ms
       
       7 packages are looking for funding
         run `npm fund` for details
       
       found 0 vulnerabilities
       npm notice 
       npm notice New minor version of npm available! 9.5.1 -> 9.8.0
       npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.8.0>
       npm notice Run `npm install -g npm@9.8.0` to update!
       npm notice 
       
-----> Build succeeded!
-----> Python app detected
-----> No Python version was specified. Using the same version as the last build: python-3.11.4
       To use a different version, see: https://devcenter.heroku.com/articles/python-runtimes
-----> No change in requirements detected, installing from cache
-----> Using cached install of python-3.11.4
-----> Installing pip 23.1.2, setuptools 67.8.0 and wheel 0.40.0
-----> Installing SQLite3
-----> Installing requirements with pip
       Collecting yt-dlp@ https://github.com/yt-dlp/yt-dlp/archive/master.tar.gz (from -r requirements.txt (line 15))
         Downloading https://github.com/yt-dlp/yt-dlp/archive/master.tar.gz (2.4 MB)
         Installing build dependencies: started
         Installing build dependencies: finished with status 'done'
         Getting requirements to build wheel: started
         Getting requirements to build wheel: finished with status 'done'
         Installing backend dependencies: started
         Installing backend dependencies: finished with status 'done'
         Preparing metadata (pyproject.toml): started
         Preparing metadata (pyproject.toml): finished with status 'done'
-----> Discovering process types
       Procfile declares types -> web
-----> Compressing...
       Done: 180.1M
-----> Launching...
       Released v34
       https://better-bard-2e85694c8cde.herokuapp.com/ deployed to Heroku
```

**Comment by mischano** [comment]
Edit: you can go to deployed website on: https://better-bard-2e85694c8cde.herokuapp.com/ and type `play robbie williams feel`, then if you open the developers console you will see the m3u8 link that I'm console logging.

**Comment by bashonly** [comment]
it's because of the iOS formats that are being extracted now

You can do this with your params:
```diff
     `ydl_opts = {
-        'format': 'bestaudio/best',
+        'format': '(bestaudio/best)[protocol~="^https?$"]',
         'quiet': True,
         'postprocessors': [{
             'key': 'FFmpegExtractAudio',
             'preferredcodec': 'mp3',
             'preferredquality': '192',
         }]
     }
```

**Comment by mischano** [comment]
@bashonly thank you for your response. I made the suggested change, but `extract_info` is still extracting in `.m3u8` format.  

I noticed my Heroku build log says `Variable FFMPEG_DOWNLOAD_URL isn't set, using default value` - could it be because of this? 

Other than that, I can't think of why this would happen. Again, the log says `Successfully built ffmpeg yt-dlp` so there should be no errors associated on Heroku side I think...
